postgres-types/src/jiff_01.rs

```diff
@@ -0,0 +1,118 @@
+use bytes::BytesMut;
+use jiff_01::{
+    civil::{Date, DateTime, Time},
+    tz::TimeZone,
+    Span, Timestamp as JiffTimestamp, Zoned,
+};
+use postgres_protocol::types;
+use std::error::Error;
+
+use crate::{FromSql, IsNull, ToSql, Type};
+
+const fn base() -> DateTime {
+    DateTime::constant(2000, 1, 1, 0, 0, 0, 0)
+}
+
+impl<'a> FromSql<'a> for DateTime {
+    fn from_sql(_: &Type, raw: &[u8]) -> Result<DateTime, Box<dyn Error + Sync + Send>> {
+        let t = types::timestamp_from_sql(raw)?;
+        Ok(base().checked_add(Span::new().microseconds(t))?)
+    }
+
+    accepts!(TIMESTAMP);
+}
+
+impl ToSql for DateTime {
+    fn to_sql(&self, _: &Type, w: &mut BytesMut) -> Result<IsNull, Box<dyn Error + Sync + Send>> {
+        types::timestamp_to_sql(self.since(base())?.get_microseconds(), w);
+        Ok(IsNull::No)
+    }
+
+    accepts!(TIMESTAMP);
+    to_sql_checked!();
+}
+
+impl<'a> FromSql<'a> for JiffTimestamp {
+    fn from_sql(type_: &Type, raw: &[u8]) -> Result<JiffTimestamp, Box<dyn Error + Sync + Send>> {
+        Ok(DateTime::from_sql(type_, raw)?
+            .to_zoned(TimeZone::UTC)?
+            .timestamp())
```

Comment:
I'm not intricately familiar with how PostgreSQL represents time, but it seems like you can avoid going through a `DateTime` here and just compute a `JiffTimestamp` directly.

Reply:
Addressed in [f00d208](https://github.com/sfackler/rust-postgres/pull/1164/commits/f00d208959c8c76c9bbe943f53a9a261ef1d2315)